### PR TITLE
Fix keptn multistage qualitygates tutorial

### LINKED ISSUE #186: Typo in tutorial keptn-multistage-qualitygates
Hello, on this page : https://tutorials.keptn.sh/tutorials/keptn-multistage-qualitygates-09/index.html?index=..%2F..index#9
the commands to add ressources are bad

We created a service named **helloserver**, and in the `keptn add-resource` we specify a service named **helloservice**

Bad:
```
keptn add-resource --project=pod-tato-head --stage=hardening --service=helloservice --resource=jmeter/load.jmx --resourceUri=jmeter/load.jmx
keptn add-resource --project=pod-tato-head --stage=hardening --service=helloservice --resource=jmeter/jmeter.conf.yaml --resourceUri=jmeter/jmeter.conf.yaml
```

Good:
```
keptn add-resource --project=pod-tato-head --stage=hardening --service=helloserver --resource=jmeter/load.jmx --resourceUri=jmeter/load.jmx
keptn add-resource --project=pod-tato-head --stage=hardening --service=helloserver --resource=jmeter/jmeter.conf.yaml --resourceUri=jmeter/jmeter.conf.yaml
```

### PR DESCRIPTION
**Pull request by TannerGabriel**
Fixing multiple errors in the Keptn multistage quality gates tutorial and switching to a revised version of the podtatohead in the `keptn/examples` repository to avoid version conflicts with the tutorial and the podatohead project.

## Fixes 
#186

### CODE REVIEW

File: site/tutorials/keptn-multistage-qualitygates-09.md [hunk 1]
```diff
@@ -128,31 +128,38 @@ In the `shipyard.yaml` shown above, we define two stages called *hardening* and
...
 
 Now each time Keptn triggers the test execution, the JMeter service will pick up both files and execute the tests.
 
+Lastly before deploying our first build we also need to define the endpoint the requests should be made against:
+
+<!-- command -->
+```
+keptn add-resource --project=podtatohead --service=helloservice --stage=hardening --resource=helm/endpoints.yaml --resourceUri=helm/endpoints.yaml
+```
+
```
Comment (christian-kreuzberger-dtx) [inline site/tutorials/keptn-multistage-qualitygates-09.md:162]: Why is this needed? It's only used for stage hardening and only contains
```
deploymentURIsLocal:
  - "http://helloservice.podtatohead-hardening:80" 
```
Can you please verify that the tutorial works without it?

File: site/tutorials/keptn-multistage-qualitygates-09.md [hunk 2]
```diff
@@ -162,12 +169,12 @@ We are now ready to kick off a new deployment of our test application with Keptn
 
     <!-- command -->
     ```
-    keptn trigger delivery --project="pod-tato-head" --service=helloservice --image="gabrieltanner/hello-server" --tag=v0.1.1
+    keptn trigger delivery --project="podtatohead" --service=helloservice --image="docker.io/jetzlstorfer/helloserver" --tag=0.1.1
```
Comment (christian-kreuzberger-dtx) [inline site/tutorials/keptn-multistage-qualitygates-09.md:172]: @johannes-b Not sure if it's a good idea that we are hosting an image used in our tutorials on a "private" account on DockerHub that we can not access. Do we have any alternatives available?

File: site/tutorials/keptn-multistage-qualitygates-09.md [hunk 3]
```diff
@@ -49,14 +49,14 @@ The demo resources can be found on Github for a convenient experience. Let's clo
 
 <!-- command -->
 ```
-git clone https://github.com/cncf/podtato-head.git
+git clone https://github.com/keptn/examples --single-branch
```
Comment (christian-kreuzberger-dtx) [inline site/tutorials/keptn-multistage-qualitygates-09.md:52]: Please always refer to a release tag/branch, see https://github.com/keptn/tutorials/blob/master/site/tutorials/snippets/09/manage/createProject.md as an example.

Using the default branch of the keptn/examples repo introduces a risk that we will have an incompatible change at some point.

**Review by christian-kreuzberger-dtx — CHANGES_REQUESTED** [review]
In general I like the changes made to the tutorial. I've left some comments.

@TannerGabriel @johannes-b I'm aware that this PR copies the steps from the Quickstart in keptn/examples, so I'm fine with just using those steps as they are, but let's discuss.

File: site/tutorials/keptn-multistage-qualitygates-09.md (hunk 1 above)
Comment (christian-kreuzberger-dtx) [reply site/tutorials/keptn-multistage-qualitygates-09.md:162]: Just got information that this is required for user-managed helm charts.

However, I believe we should consider adding an endpoints.yaml file for all stages - see https://github.com/keptn/examples/issues/201

File: site/tutorials/keptn-multistage-qualitygates-09.md (hunk 2 above)
Comment (christian-kreuzberger-dtx) [reply site/tutorials/keptn-multistage-qualitygates-09.md:172]: We've decided to move those images to a Keptn-related organization. We will update this PR with a comment soon.

File: site/tutorials/keptn-multistage-qualitygates-09.md [hunk 4]
```diff
@@ -128,31 +128,38 @@ In the `shipyard.yaml` shown above, we define two stages called *hardening* and
...
 
-After creating the project, we can continue by onboarding the *helloserver* as a service to your project using the `keptn create service` and `keptn add-resource` commands. You need to pass the project where you want to create the service, as well as the Helm chart of the service.
-For this purpose we need the helm charts as a tar.gz archive. To archive it use following command:
+After creating the project, we can continue by onboarding the *helloservice* as a service to your project using the `keptn create service` and `keptn add-resource` commands. You need to pass the project where you want to create the service, as well as the Helm chart of the service.
+For this purpose we need the helm charts as a tar.gz archive. Our repository already contains a tar.gz archive of our helm chart but normally you could use the following command for the conversion:
 
 <!-- command -->
 ```
-tar cfvz ./helm-charts/helloserver.tgz ./helm-charts/helloserver
+tar cfvz ./helm/helloservice.tgz ./helm/helloservice
```
Comment (christian-kreuzberger-dtx) [inline site/tutorials/keptn-multistage-qualitygates-09.md:136]: Either we remove the .tgz file from https://github.com/keptn/examples/tree/master/quickstart/demo/helm or we remove the step from this tutorial. But having both is redundant and might confuse the user.

In order to be in-line with the other tutorials, we should remove the step from the tutorial.

Comment (TannerGabriel) [reply site/tutorials/keptn-multistage-qualitygates-09.md:136]: That is fine. I will remove it in the next commit.

File: site/tutorials/keptn-multistage-qualitygates-09.md (hunk 2 above)
Comment (TannerGabriel) [reply site/tutorials/keptn-multistage-qualitygates-09.md:165]: Let me know when the new Docker image is available then I will change the image all over the tutorial.

File: site/tutorials/keptn-multistage-qualitygates-09.md [hunk 5]
```diff
@@ -128,31 +128,31 @@ In the `shipyard.yaml` shown above, we define two stages called *hardening* and
...
-After onboarding the service, tests (i.e., functional- and performance tests) need to be added as basis for quality gates. We are using JMeter tests, as the JMeter service comes "batteries included" with our Keptn installation.
+Now each time Keptn triggers the test execution, the JMeter service will pick up both files and execute the tests.
+
+Lastly before deploying our first build we also need to define the endpoint the requests should be made against:
 
 <!-- command -->
 ```
-keptn add-resource --project=pod-tato-head --stage=hardening --service=helloservice --resource=jmeter/load.jmx --resourceUri=jmeter/load.jmx
-keptn add-resource --project=pod-tato-head --stage=hardening --service=helloservice --resource=jmeter/jmeter.conf.yaml --resourceUri=jmeter/jmeter.conf.yaml
+keptn add-resource --project=podtatohead --service=helloservice --all-stages --resource=helm/endpoints.yaml --resourceUri=helm/endpoints.yaml
```
Comment (TannerGabriel) [inline site/tutorials/keptn-multistage-qualitygates-09.md:153]: This needs to be changed to use different endpoint files for the single stages after the merge of [this PR](https://github.com/keptn/examples/pull/205).

File: site/tutorials/keptn-multistage-qualitygates-09.md (hunk 2 above)
Comment (TannerGabriel) [reply site/tutorials/keptn-multistage-qualitygates-09.md:166]: @christian-kreuzberger-dtx Any progress here?

Comment (christian-kreuzberger-dtx) [reply site/tutorials/keptn-multistage-qualitygates-09.md:166]: This is planned for one of the next sprints, but no progress yet.

Comment (christian-kreuzberger-dtx) [reply site/tutorials/keptn-multistage-qualitygates-09.md:166]: We have finished the changes on the quickstart, please use the following images and tags:

* normal version: `ghcr.io/podtato-head/podtatoserver` - `v0.1.1` (don't forget to use the prefix `v`)
* slow version: `ghcr.io/podtato-head/podtatoserver` - `v0.1.2` (don't forget to use the prefix `v`)

Comment (TannerGabriel) [reply site/tutorials/keptn-multistage-qualitygates-09.md]: It should be resolved now!

**Comment by TannerGabriel** [comment]
@christian-kreuzberger-dtx The only problem we have right now is that the slow build delay of the [official podtato-head](https://github.com/podtato-head/podtato-head/blob/main/podtato-server/podtatoserver.go#L84) is 2 seconds and not the 400ms we are using in the SLO file. I will submit a PR explaining why 400ms is more suitable for running tests (shorter wait times on failing builds).

**Comment by christian-kreuzberger-dtx** [comment]
@TannerGabriel Thanks for the changes. The delay of 2s is fine, I've adapted the SLO and SLI files accordingly.

**Comment by oleg-nenashev** [comment]
IMHO we should merge this PR and then create similar fixes for the upstream

**Review by christian-kreuzberger-dtx — APPROVED** [review]
LGTM

**Review by oleg-nenashev — APPROVED** [review]
I created https://github.com/keptn/tutorials/issues/222 as a follow-up
